deploy/ods-pipeline/charts/tasks/templates/task-ods-build-gradle.yaml

```diff
@@ -50,6 +50,21 @@ spec:
         You must set this to `"false"` if the build can be affected by files outside `working-dir`. See ADR caching-build-tasks for more details and workarounds.
       type: string
       default: "true"
+    - name: cache-extra-inputs
+      description: |
+        Paths in addition to the working-dir which are used as inputs by the build.
+        If there are any changes in these locations a prior cache can no longer be used and thus the build task will rebuild from scratch.
+        The paths are relative to the repository root and separated by colon.
+        If caching is not enabled this value is not used.
+      type: string
+      default: ""
+    - name: cached-outputs
+      description: >-
+        List of build output directories (as colon separated string) to be cached.
+        These directories are relative to `working-dir`.
+        If caching is not enabled this value is not used.
```

Comment:
I think we do not need the separate toggle `cache-build` anymore. Its description is a bit outdated now, and I think we could have the caching be based on the presence of the `cached-outputs` param. If it is the empty string, no caching should happen. If one or more paths are specified, those should be cached.

That may lead to the question what we should have as defaults for this param, and I am leaning towards using the empty string as a default, and making build caching opt in. I feel that without an explicit output-dir, caching is more subtle and it would be good if people read about the feature and understand what has to be done instead of unknowingly using some magic values which may or may not cache everything they need.

Reply:
Finally I decided to remove extra inputs and instead have a `cache-sources` parameter:

- List of build source directories (as colon separated string) which influence the build.
  These directories are relative to the repository root.
   If the contents in these directories change the cache is invalidated so that the build task will rebuild from scratch.
  **If left empty no caching is enabled**. When sources are only in the working-dir, this value can be the same value.

You will have to opt in to build task skipping using parameter `cache-sources` for all languages. 

The `cache-outputs` parameter can be used to determine the folders inside `working-dir` to cache which is used as follows in the language tasks:

- go: not supported, passed into caching scripts with value of `output-dir` 
- npm: supported with default of `'dist'`
- gradle:  supported with default of `'docker'`
- python: not supported, passed into caching scripts as empty value. As a reminder a build may still be skipped if the input doesn't change, but there is no need to cache any built files.

To me this feels quite natural now. What do you think?